src/shared/conf-parser.c

```diff
@@ -1185,3 +1185,35 @@ int config_parse_permille(const char* unit,
 
         return 0;
 }
+
+int config_parse_vlanprotocol(const char* unit,
+                              const char *filename,
+                              unsigned line,
+                              const char *section,
+                              unsigned section_line,
+                              const char *lvalue,
+                              int ltype,
+                              const char *rvalue,
+                              void *data,
+                              void *userdata) {
+        int *vlanprotocol = data;
```

Comment:
it's a bit weird to name the field in the structure `vlan_protocol` and the local one `vlan_protocol`. I mean, it's fine if the local var is named differently, but different in the underscore is strange. i.e. maybe name this `vp` or use the same `vlan_protocol` spelling as in the structure, but this is just weird...